addons/stock/tests/test_report.py

```diff
@@ -39,22 +48,43 @@ def get_report_forecast(self, product_template_ids=False, product_variant_ids=Fa
 
 
 class TestReports(TestReportsCommon):
-    def test_reports(self):
-        product1 = self.env['product.product'].create({
-            'name': 'Mellohi',
-            'default_code': 'C418',
+
+    def test_product_label_reports(self):
+        report = self.env.ref('stock.label_product_product')
+        target = b'\n\t\t\n\n\n\n\n\n^XA^CI28\n^FT100,80^A0N,40,30^FD[C4181234""154654654654]Mellohi"^FS\n\n^FT100,115^A0N,30,24^FDC4181234""15465^FS\n^FT100,150^A0N,30,24^FD4654654^FS\n\n\n\n^FO100,160^BY3\n^BCN,100,Y,N,N\n^FDscan""me^FS\n\n^XZ\n\n\n\n^XA^CI28\n^FT100,80^A0N,40,30^FD[C4181234""154654654654]Mellohi"^FS\n\n^FT100,115^A0N,30,24^FDC4181234""15465^FS\n^FT100,150^A0N,30,24^FD4654654^FS\n\n\n\n^FO100,160^BY3\n^BCN,100,Y,N,N\n^FDscan""me^FS\n\n^XZ\n\n\n\n\n'
+        rendering, qweb_type = report._render_qweb_text(self.product1.product_tmpl_id.id, {'quantity_by_product': {self.product1.product_tmpl_id.id: 2}, 'active_model': 'product.template'})
+        self.assertEqual(target, rendering.replace(b' ', b''), 'The rendering is not good')
+        self.assertEqual(qweb_type, 'text', 'the report type is not good')
+
+    def test_product_label_without_default_code_reports(self):
```

Comment:
Not sure I understand the purpose of this test ? Is it solely to check: 
https://github.com/odoo/odoo/blob/61c0965d1de8d185a7c7115a36433ce3a3b455fd/addons/stock/report/product_label_report.py#L26
Because if that's the case, not sure it warrants a test. :slightly_smiling_face: